### DISCUSSION TITLE
Delete Operations are too heavy/slow for ajax

### DISCUSSION THREAD

**Discussion by timkelty** [craftcms/cms#8794] [Ideas]
I have a site where I can't delete things, because they try to execute in an ajax call, and there is just too much data that it never succeeds (timeout).

For example:

- deleting sections
- deleting fields
- any kind of bulk element delete (checking boxes and using element action -> Delete)

Ultimately I think things should be flagged for deletion, and then deleted via queue job.
…but an easier short-term solution would be to just provide console controllers so this could be done via cli.

**Comment by timkelty** [comment]
FWIW, I've had the same trouble with deleting sections,  as it seems to go through and delete every entry which takes…forever.

**Comment by jsunsawyer** [comment]
I think I just ran into this as well. In my case, I was bulk deleting some test assets from a staging environment (about 20-30 image assets).

The assets in the CP fade out, and it's hard to tell if things are still working or if it has stalled out. After some time, refreshing the page will show that some of the selected assets were deleted, but some remain.

When these types of operations last for too long (minutes or close?), it makes the CP feel broken/stalled.
